### ISSUE TITLE
Rule Change: Adopt _binding convention to mark unused variables

### ISSUE THREAD

**Issue by bugsn4x** [eslint/eslint#15429]
### What rule do you want to change?

unused variables

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
Many programming languages have a convention of denoting unused variables, by prefixing the variable name with a single underscore. Like `_e`, `_i`, or `_collection`.

This is useful for communicating to contributors the intention to ignore certain unused parameters. For example, when a callback is required to match a signature accepting many arguments, but the implementation does not need to interact with all the arguments. Hence the underscore convention used in many programming languages.

Rest parameters may serve as a workaround, except in cases where legacy Internet Explorer support is required.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters
```

### What does the rule currently do for this code?

Ignores the underscore convention and continues to report variables like `_e` that are not used.

### What will the rule do after it's changed?

Silence warnings for variables that are correctly marked unused via the underscore convention.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @bugsn4x, thanks for the issue!

> Silence warnings for variables that are correctly marked unused via the underscore convention.

This is already supported by options [`varsIgnorePattern`](https://eslint.org/docs/rules/no-unused-vars#varsignorepattern) and [`argsIgnorePattern`](https://eslint.org/docs/rules/no-unused-vars#argsignorepattern).

If you want to ignore unused parameters prefixed with an underscore, you can configure the `no-unused-vars` rule like this:

```js
no-unused-vars: ["error", { "argsIgnorePattern": "^_" }]
```

An example with no linting errors:

```js
/* eslint no-unused-vars: ["error", { "argsIgnorePattern": "^_" }] */

(_unused) => {}
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiBbXCJlcnJvclwiLCB7IFwiYXJnc0lnbm9yZVBhdHRlcm5cIjogXCJeX1wiIH1dICovXG5cbihfdW51c2VkKSA9PiB7fVxuICAiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

If you additionally want to ignore other kinds of variables (not just parameters) prefixed with an underscore, you can also set `"varsIgnorePattern": "^_" `:

```js
/* eslint no-unused-vars: ["error", { "argsIgnorePattern": "^_", "varsIgnorePattern": "^_" }] */

(_unused) => {}

let _unused;
```

**Comment by bugsn4x** [comment]
Ah, that is a really helpful option!

Would be useful to make this default behavior.

**Comment by mdjermanovic** [comment]
> Would be useful to make this default behavior.

I don't think this rule should imply any naming conventions. It's fairly easy to configure the options to enforce the underscore or any other convention. Also, changing the default behavior would be a big breaking change now.
